packages/core/http/core-http-server/src/router/router.ts

```diff
@@ -86,11 +86,7 @@ export interface IRouter<Context extends RequestHandlerContextBase = RequestHand
    * @internal
    */
   getRoutes: () => RouterRoute[];
-}
 
-export interface IRouterWithVersion<
-  Context extends RequestHandlerContextBase = RequestHandlerContextBase
-> extends IRouter<Context> {
```

Comment:
This is the main change of this PR: just moved the `versioned` property to the `IRouter` interface and removed the `IRouterWithVersion` interface entirely.

Reply:
Just to be clear: will `version` remain optional until we implement versioning on self-managed too?
Some plugins might take the option of implementing _new_ routes for serverless and keep the existing ones as-is.

If we enforce `version` on all routes, it will be a breaking change for public APIs that already exist.

Reply:
This change only adds the `versioned` property to the `IRouter` interface and leaves existing route definitions unchanged. The name `IRouterWithVersion` is a bit confusing (just another reason to get rid of it haha).